src/frontend/screens/SendJoinRequestScreen.tsx

```diff
@@ -0,0 +1,126 @@
+import * as React from "react";
+import { Share, StyleSheet, View } from "react-native";
+import { FormattedMessage, defineMessages } from "react-intl";
+import { getUniqueId } from "react-native-device-info";
+import {
+  NavigationStackScreenComponent,
+  TransitionPresets,
+} from "react-navigation-stack";
+
+import useWifiStatus from "../hooks/useWifiStatus";
+import { MEDIUM_BLUE, WHITE } from "../lib/styles";
+import HeaderTitle from "../sharedComponents/HeaderTitle";
+import { BackIcon } from "../sharedComponents/icons";
+import Button from "../sharedComponents/Button";
+import IconButton from "../sharedComponents/IconButton";
+import WifiBar from "../sharedComponents/WifiBar";
+import Text from "../sharedComponents/Text";
+import useProjectInviteListener from "../hooks/useProjectInviteListener";
+import { URI_PREFIX } from "../constants";
+
+const m = defineMessages({
+  title: {
+    id: "screens.SendJoinRequest.title",
+    defaultMessage: "Send Join Request",
+  },
+  verificationCode: {
+    id: "screens.SendJoinRequest.verificationCode",
+    defaultMessage: "Verification Code",
+  },
+  instructions: {
+    id: "screens.SendJoinRequest.instructions",
+    defaultMessage: "Share this code with the Project Coordinator",
+  },
+  share: {
+    id: "screens.SendJoinRequest.share",
+    defaultMessage: "Share",
+  },
+});
+
+const SendJoinRequestScreen: NavigationStackScreenComponent = () => {
+  useProjectInviteListener();
+
+  const { ssid } = useWifiStatus();
+
+  const deviceName = "Android " + getUniqueId().slice(0, 4).toUpperCase();
+
+  // TOOD: Need to properly generate
+  const verificationCode = Math.random().toString().slice(-5);
+
+  // TODO: Decide on a URL structure for this deep link
+  const shareLink = `${URI_PREFIX}main/onboarding?code=${verificationCode}`;
+
+  return (
+    <View style={styles.pageContainer}>
+      <WifiBar deviceName={deviceName} ssid={ssid} />
+      <View style={styles.container}>
+        <View>
+          <Text style={styles.verificationCodeTitle}>
+            <FormattedMessage {...m.verificationCode} />
+          </Text>
+          <Text style={styles.verificationCode}>{verificationCode}</Text>
+          <Text style={styles.instructions}>
+            <FormattedMessage {...m.instructions} />
+          </Text>
+        </View>
+        <Button
+          onPress={() => Share.share({ message: shareLink, url: shareLink })}
+        >
+          <Text style={styles.shareButton}>
+            <FormattedMessage {...m.share} />
+          </Text>
+        </Button>
+      </View>
+    </View>
+  );
+};
+
+SendJoinRequestScreen.navigationOptions = () => ({
+  ...TransitionPresets.SlideFromRightIOS,
```

Comment:
although it's more of an iOS pattern, we have this screen sliding in from the right side - felt more natural and intuitive than from the bottom